core/cap-memo.md

```diff
@@ -0,0 +1,106 @@
+```
+CAP: ????
+Title: Smart Contract Standardized Asset (Stellar Asset Contract) Extension: Memo
+Working Group:
+    Owner: Tomer Weller <@tomerweller>
+    Authors: Leigh McCulloch <@leighmcculloch>, Tomer Weller <@tomerweller>
+    Consulted: Nicolas Barry <@monsieurnicolas>, Alex Cordeiro <@accordeiro>, Dmytro Kozhevin <@dmkozh>
+Status: Draft
+Created: 2024-08-26
+Discussion: TBA
+```
+
+## Simple Summary
+
+Implement the [SEP-memo] for the Stellar Asset Contract.
+
+## Motivation
+
+To support an unambiguous method by which contract wallets can transfer Stellar Assets to custodial wallets that require
+a memo to be associated with the transfer that is populated through to events used to track transfers.
+
+### Goals Alignment
+
+This CAP is aligned with the following Stellar Network Goals:
+
+- The Stellar Network should make it easy for developers of Stellar projects to create highly usable products
+- The Stellar Network should enable cross-border payments, i.e. payments via exchange of assets, throughout the globe,
+  enabling users to make payments between assets in a manner that is fast, cheap, and highly usable.
+
+## Abstract
+
+This proposal introduces a built-in implementation for Stellar Assets of the [SEP-memo] standard for transferring with
+an associated memo.
+
+This proposal disallows the subsequent use of the transaction memo field on smart contract transactions.
+
+## Specification
+
+### XDR Changes
+
+None.
+
+### Transactions Changes
+
+The `memo` field of `Transaction` must have value `MEMO_NONE` when `SorobanTransactionData` is present.
+
+### Extended Token Interface
+
+The Stellar Asset Contract interface is extended with one function:
+
+```rust
+/// Transfer `amount` from `from` to `to`.
+///
+/// The memo annotates the transfer event.
+///
+/// # Arguments
+///
+/// - `from` - The address holding the balance of tokens which will be
+/// withdrawn from.
+/// - `to` - The address which will receive the transferred tokens.
+/// - `amount` - The amount of tokens to be transferred.
+/// - `memo` - The memo associated with the transfer.
+///
+/// # Events
+///
+/// Emits an event with topics `["transfer", from: Address, to: Address, memo: u64, sep0011_asset: String],
+/// data = amount: i128`
+fn transfer_memo(env: Env, from: Address, to: Address, amount: i128, memo: u64);
```

Comment:
How about calling the field memo_id?

Reply:
https://github.com/stellar/stellar-protocol/discussions/1534